kuzzle-sdk-java/src/main/java/io/kuzzle/sdk/Kuzzle.java

```diff
@@ -0,0 +1,284 @@
+     * If set to -1 the SDK does not refresh the token automatically.
+     */
+    protected AtomicInteger minTokenDuration;
+
+
+    /**
+     * The minimum duration of a Token after refresh.
+     * If set to -1 the SDK does not refresh the token automatically.
+     */
+    protected AtomicInteger refreshedTokenDuration;
+
+    /**
+     * The maximum delay between two requests to be replayed
+     */
+    protected AtomicInteger maxRequestDelay;
+
+    protected ConcurrentHashMap<String, Task<Response>>
+                    requests = new ConcurrentHashMap<>();
+
+    /** Initialize a new instance of Kuzzle
+     * @param networkProtocol   The network protocol
+     * @throws IllegalArgumentException
+     */
+    public Kuzzle(AbstractProtocol networkProtocol)
+            throws IllegalArgumentException {
+        this(networkProtocol, new KuzzleOptions());
+    }
+
+    /** Initialize a new instance of Kuzzle
+     * @param networkProtocol   The network protocol
+     * @param options           Kuzzle options
+     * @throws IllegalArgumentException
+     */
+    public Kuzzle(
+            AbstractProtocol networkProtocol,
+            final KuzzleOptions options
+    ) throws IllegalArgumentException {
+
+        if (networkProtocol == null) {
+            throw new IllegalArgumentException("networkProtocol can't be null");
+        }
+
+        KuzzleOptions kOptions = options != null
+                ? options
+                : new KuzzleOptions();
+
+        this.networkProtocol = networkProtocol;
+        this.networkProtocol.registerResponseEvent(this::onResponseReceived);
+        this.networkProtocol.registerStateChangeEvent(this::onStateChanged);
+
+        this.maxQueueSize = new AtomicInteger(kOptions.getMaxQueueSize());
+        this.minTokenDuration = new AtomicInteger(kOptions.getMinTokenDuration());
+        this.refreshedTokenDuration = new AtomicInteger(kOptions.getRefreshedTokenDuration());
+        this.maxRequestDelay = new AtomicInteger(kOptions.getMaxRequestDelay());
+
+        this.version = "3.0.0";
+        this.instanceId = UUID.randomUUID().toString();
+
+        this.tokenExpiredEvent = new EventListener();
+        this.unhandledResponseEvent = new EventListener<>();
+    }
+
+    /** Establish a network connection
+     * @throws Exception
+     */
+    public void connect() throws Exception {
+        networkProtocol.connect();
+    }
+
+    /**
+     * Disconnect this instance
+     */
+    public void disconnect() {
+        networkProtocol.disconnect();
+    }
+
+    /** Handles the ResponseReceivedEvent from the network protocol
+     * @param payload Raw API Response
+     */
+    protected void onResponseReceived(String payload) {
+
+        Response response = new Response();
+        response.fromMap(JsonSerializer.deserialize(payload));
+
+        if (response.room != null
+            && requests.containsKey(notNull(response.room, ""))
```

Comment:
Can't use set a default value for `room` inside the `Response` object instead of using this helper?
(apply for the other use below)

Reply:
No because `room` can be null but I should have changed it to
```suggestion
            && requests.containsKey(response.room)
```
Just because response.room is a non null string in this case